source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,255 @@
+- CPU or memory oversubscription due to connection management can lead
+  to a ``mongod`` crashing and restarting. This extra workload on the server
+  can be exclusively attributed to connection handling.
+- In Atlas, connection requests can exceed instance class limits, causing
+  refusal of new connections.
+
+Avoid Connection Storms
+-----------------------
+
+While no universal solution exists as each problem is unique to a deployment,
+you can use a combination of settings and methods to help you avoid connection
+storms. These include:
+
+- **maxConnecting**: The maximum number of connections the driver can concurrently
+  establish to an individual server before making them available in the connection pool.
+  |
+  By keeping this option low, you reduce the risk temporary load spikes
+  from  overwhelming the database.
+
+- **maxPoolSize**: The maximum number of connections in the connection pool. Beginning
+  with a lower value is recommended to reduce the maximum pressure each server
+  may experience from each connected client applications.
+  |
+  Fewer overall connections in the pool means the database server has
+  is using less resources maintaining connections.
+
+- **maxTime()**: The *cumulative* time limit for processing operations on
+  a MongoDB server.
+  |
+  Terminating long-running operations frees up resources on the server and
+  hints that there may be underlying performance issues (like a lack of appropriate indexes).
+
+maxConnecting
++++++++++++++
+
+``maxConnecting`` specifies the maximum number of connections to each server that
+the driver will concurrently establish before making them available in
+the connection pool. The default is 2.
+
+.. important:: Start with Lower Values
+
+   Start with lower values for ``maxConnecting``. This helps to avoid bursts
+   in connections in response to a temporary increase in traffic.
+
+
+Do not confuse ``maxConnecting`` with ``maxPoolSize``, which specifies the
+maximum number of connections the driver can maintain in the connection pool.
+
+.. example::
+
+   Consider a failover scenario in a 3 member replica set with applications under
+   heavy load. Without ``maxConnecting``, ``maxPoolSize * application servers``
+   connections would be established. With default settings for ``maxPoolSize``
+   and 3 application servers, that would be 300 new connections, concurrently.
+
+   With ``maxConnecting`` set to 3, the new primary will receive
+   ``maxConnecting * application servers`` connections concurrently, or 9.
+
+Specify ``maxConnecting`` in your connection URI or with the
+`ConnectionPoolSettings.Builder maxConnecting() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ConnectionPoolSettings.Builder.html#maxConnecting(int)>`__ method.
+
+maxPoolSize
++++++++++++
+
+:ref:`Properly sizing your connection pool <connection-pool-sizing-java>`
+helps you avoid the risk of connection storms by limiting the amount of
+connections any application can establish.
```

Comment:
Sometimes the alternation between passive and imperative language makes it difficult to locate the specific advice. Maybe the imperative style is better since we try to avoid passive voice when possible:
"Properly size your connection pool to avoid ..."